src/Geometry.cpp

```diff
@@ -16,510 +14,744 @@
-    std::cout << " " << std::endl;
-  }
-
-  // construct full hexagon cells
-  for (unsigned int icell=0; icell<full_hexagons_count.asUInt(); icell++) {
-    std::string cell_name = fixedLength(icell,5,"FH");
-    // FIXME: add more checks
-    const Json::Value& hexagon_attributes = hexagons[cell_name]; 
-    int i_index = hexagon_attributes[0]["mapping_coord"][0].asInt();
-    int j_index = hexagon_attributes[0]["mapping_coord"][1].asInt();
-    TVectorD position(3);
-    position(0) = hexagon_attributes[1]["center_coord"][0].asDouble()/10.; // cm
-    position(1) = hexagon_attributes[1]["center_coord"][1].asDouble()/10.; // cm
-    position(2) = getZlayer();
-    double orientation = hexagon_attributes[2]["orientation"].asDouble();
-    std::vector<TVectorD> vertices;
-    for (unsigned i=0; i<hexagon_attributes[3]["vertex_coord_abs"].size(); i++) {
-      vertices.emplace_back(2);
-      vertices.back()(0) = hexagon_attributes[3]["vertex_coord_abs"][i][0].asDouble()/10.; // cm
-      vertices.back()(1) = hexagon_attributes[3]["vertex_coord_abs"][i][1].asDouble()/10.; // cm
+    reverse(result.begin(), result.end());
+    result = type + result;
+    return result;
+}
+
 
+void Geometry::setLayer(int klayer) {
+    if (klayer<0 || unsigned(klayer)>=parameters_.layers_z.size()) {
+        stringstream error;
+        error << "[Geometry] error, invalid klayer " << klayer;
+        throw error.str();
     }
-    cells_.emplace(
-        Cell::id(i_index, j_index),
-        Cell(std::move(position), std::move(vertices), orientation, i_index, j_index)
-        );
+    klayer_ = klayer;
+}
 
-    if(debug) {
-      std::cout << "New cell " << cell_name << " : " << std::endl;
-      std::cout << " mapping coordinates : " << 
-        hexagon_attributes[0]["mapping_coord"][0].asInt() << " " <<
-        hexagon_attributes[0]["mapping_coord"][1].asInt() << std::endl;
-      std::cout << " center coordinates : " << 
-        hexagon_attributes[1]["center_coord"][0].asDouble() << " " <<
-        hexagon_attributes[1]["center_coord"][1].asDouble() << std::endl;
-      std::cout << " orientation : " << 
-        hexagon_attributes[2]["orientation"].asDouble() << std::endl;
-      std::cout << " vertex_coordinates : " << std::endl;
-      for(const auto& vertex : vertices) {
-        std::cout << "  vertex " << 
-          vertex(0) << " " <<
-          vertex(1) << std::endl;
-      }
+
+string Geometry::setHgcalPart(int klayer) {
+    if (klayer < 28)
+        return "EE";
+    else
+        return "FH";
+}
+
+double* Geometry::dimensions(double side) {
+    double *dim = new double[4];
+
+    dim[0] = side*sqrt(3.); //asqrt3_
+    dim[1] = dim[0]/2.;  //asqrt3over2_
+    dim[2] = side/2.;          //aover2_
+    dim[3] = dim[2]*3.;      //a3over2
+
+    return dim;
+}
+
+double** Geometry::hexagonoffset(double side) {
+    // pic on top
+    double *dim = dimensions(side);
+    return new double*[2] {
```

Comment:
Here and everywhere else in this file. Use variables with meaningful names instead of the magic numbers used for the array sizes.

Reply:
Fixed